src/main/java/org/folio/service/inventory/InventoryManager.java

```diff
@@ -1163,9 +1165,14 @@ private Future<String> createItemInInventory(JsonObject itemData, RequestContext
     restClient.postJsonObjectAndGetId(requestEntry, itemData, requestContext)
       .onSuccess(promise::complete)
       // In case item creation failed, return null instead of id
-      .onFailure(throwable -> {
-        logger.error(ITEM_CREATION_FAILED.getDescription());
-        promise.complete();
+      .onFailure(t -> {
+        if (StringUtils.isNotEmpty(t.getMessage()) && t.getMessage().contains(BARCODE_ALREADY_EXIST_ERROR)) {
+          logger.info("Barcode is already exists, full response message: {}", t.getMessage(), t);
+          promise.fail(new HttpException(409, BARCODE_IS_NOT_UNIQUE));
```

Comment:
**Question**
Is this covered by some test?

Reply:
covered with tests